ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -370,25 +370,39 @@ data UnforwardedReadSets l = UnforwardedReadSets {
 forwardTableKeySets ::
      (TableStuff l, HasCallStack)
   => DbChangelog l
+  -> Maybe (LedgerTables l DiffMK)
   -> UnforwardedReadSets l
   -> Either (WithOrigin SlotNo, WithOrigin SlotNo)
             (LedgerTables l ValuesMK)
-forwardTableKeySets dblog = \(UnforwardedReadSets seqNo' values keys) ->
+forwardTableKeySets dblog mempoolDiffs = \(UnforwardedReadSets seqNo' values keys) ->
```

Comment:
The diffs are not necessarily mempool specific from the perspective of `forwardTableKeySets`, so maybe rename `mempoolDiffs` to something like `extraDiffs` or `auxDiffs`? Might also be good to document the parameter